tensorflow_io/core/plugins/hdfs/hadoop_filesystem.cc

```diff
@@ -0,0 +1,919 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include <stdlib.h>
+#include <string.h>
+
+#include <functional>
+#include <iostream>
+#include <map>
+#include <sstream>
+#include <string>
+#if !defined(_MSC_VER)
+#include <dlfcn.h>
+#endif
+
+#include "absl/strings/str_cat.h"
+#include "absl/synchronization/mutex.h"
+#include "hdfs/hdfs.h"
+#include "tensorflow/c/tf_status.h"
+#include "tensorflow_io/core/plugins/file_system_plugins.h"
+// TODO: Restore logging.h
+#define TF_Log(...)
+#define TF_VLog(...)
+// #include "tensorflow/c/logging.h"
+// TODO: Restore logging.h
+
+namespace tensorflow {
+namespace io {
+namespace hdfs {
+
+void* LoadSharedLibrary(const char* library_filename, TF_Status* status) {
+#if defined(_MSC_VER)
+  TF_SetStatus(status, TF_UNIMPLEMENTED, "LoadSharedLibrary not implemented");
+  return nullptr;
+#else
+  void* handle = dlopen(library_filename, RTLD_NOW | RTLD_LOCAL);
+  if (handle == nullptr) {
+    std::string error_message =
+        absl::StrCat("Library (", library_filename, ") not found: ", dlerror());
+    TF_SetStatus(status, TF_NOT_FOUND, error_message.c_str());
+    return nullptr;
+  }
+  TF_SetStatus(status, TF_OK, "");
+  return handle;
+#endif
+}
+
+void* GetSymbolFromLibrary(void* handle, const char* symbol_name,
+                           TF_Status* status) {
+#if defined(_MSC_VER)
+  TF_SetStatus(status, TF_UNIMPLEMENTED,
+               "GetSymbolFromLibrary not implemented");
+  return nullptr;
+#else
+  if (handle == nullptr) {
+    TF_SetStatus(status, TF_INVALID_ARGUMENT, "library handle cannot be null");
+    return nullptr;
+  }
+  void* symbol = dlsym(handle, symbol_name);
+  if (symbol == nullptr) {
+    std::string error_message =
+        absl::StrCat("Symbol (", symbol_name, ") not found: ", dlerror());
+    TF_SetStatus(status, TF_NOT_FOUND, error_message.c_str());
+    return nullptr;
+  }
+  TF_SetStatus(status, TF_OK, "");
+  return symbol;
+#endif
+}
+
+// Implementation of a filesystem for HADOOP environments.
+// This filesystem will support `hdfs://`, `viewfs://` and `har://` URI schemes.
+
+static void* plugin_memory_allocate(size_t size) { return calloc(1, size); }
+static void plugin_memory_free(void* ptr) { free(ptr); }
```

Comment:
These two functions are unused. Likely because I copied from `posix`. Let just merge this PR and I will make a follow up PR ( mostly cleanup and better memory management ).